core/include/scipp/core/element/util.h

```diff
@@ -24,7 +24,7 @@ template <> inline bool is_linspace(const span<const core::time_point> &range) {
   if (range.back() <= range.front())
     return false;
 
-  const auto delta = (range.back() - range.front()) / (scipp::size(range) - 1);
+  const auto delta = range.size() < 2 ? 0 : range[1] - range[0];
```

Comment:
```suggestion
  const auto delta = range[1] - range[0];
```
since you return `false` above for `size < 2`.